### PR TITLE
feat: Add performance tracking to GraphQL mutations

### PR DESCRIPTION
**Pull request by harsh04044**
**What kind of change does this PR introduce?**

Feature - Adds performance tracking instrumentation to GraphQL mutation resolvers

**Issue Number:**

Fixes #4125 

**Snapshots/Videos:**

N/A - This PR adds backend instrumentation that doesn't produce visual changes.

**If relevant, did you update the documentation?**

N/A - This is an internal performance tracking enhancement that doesn't require user-facing documentation updates.

**Summary**

This PR implements performance tracking for GraphQL mutation resolvers to enable monitoring of mutation execution times and identification of slow operations. This is part of a larger effort to add comprehensive observability to the Talawa API.

The implementation wraps five critical mutation operations (`createUser`, `createOrganization`, `createEvent`, `updateOrganization`, and `deleteOrganization`) with performance tracking using a reusable helper utility. The tracking uses the existing `PerformanceTracker` from the context, following the established pattern of `mutation:{mutationName}` for operation naming.

**Key Changes:**
- Created `executeMutation` helper utility in `src/graphql/utils/withMutationMetrics.ts` that wraps mutation logic with performance tracking
- Instrumented all five target mutations to use the new helper
- Added comprehensive test coverage for each mutation's metrics collection (both success and failure scenarios)
- Ensured graceful degradation when performance tracker is unavailable
- Maintained all existing mutation functionality without breaking changes

The implementation follows best practices:
- Uses optional chaining (`ctx.perf?.`) for safe access
- Preserves error handling and mutation behavior
- Tracks metrics even when mutations fail (important for monitoring error rates)
- Follows the established naming convention for operation tracking

**Does this PR introduce a breaking change?**

No - This is a purely additive change. All existing mutation functionality remains unchanged. The performance tracking is transparent to clients and gracefully degrades when the performance tracker is unavailable.

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Other information**

This PR is part of PR 6b (Mutation Performance Tracking) and builds upon the performance tracking infrastructure established in previous PRs. The helper utility (`executeMutation`) is designed to be reusable for instrumenting additional mutations in future PRs.

**Test Coverage Details:**
- Created dedicated `.metrics.test.ts` files for each of the five mutations
- Added comprehensive tests for the `withMutationMetrics` helper utility
- All tests verify metrics collection on both successful and failed mutation executions
- Tests cover edge cases including missing performance tracker (graceful degradation)

**Implementation Notes:**
- The helper utility uses `ctx.perf?.time()` internally to track execution time
- Operation names follow the pattern: `mutation:{mutationName}` (e.g., `mutation:createUser`)
- All mutations maintain backward compatibility and existing error handling
- No functional changes to mutation logic - only instrumentation added

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

* **Breaking Changes**
  * Removed agenda item sequence mutation and several event/agenda-related fields and query options.

* **New Features**
  * Per-mutation performance metrics for create/update/delete mutations are now recorded.

* **Improvements**
  * Stronger transactional handling and rollback for events, organizations, and users (attachments and avatars handled more safely).
  * More robust validation and error handling for create/update flows.

* **Tests**
  * Added end-to-end metrics tests and unit tests for mutation instrumentation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/graphql/types/Mutation/createEvent.ts [hunk 1]
```diff
@@ -102,259 +103,168 @@ builder.mutationField("createEvent", (t) =>
...
-								code: "unexpected",
+								code: "invalid_arguments",
+								issues: validation.errors.map((error) => ({
+									argumentPath: ["input", "recurrence"],
+									message: error,
+								})),
 							},
 						});
 					}
+				}
```
[the remark this answers is not in the document]
Comment (harsh04044) [reply src/graphql/types/Mutation/createEvent.ts:176]: @coderabbitai is this not addressed?

File: test/graphql/types/Mutation/createUser.test.ts [hunk 2]
```diff
@@ -740,4 +742,93 @@ suite("Mutation field createUser", () => {
...
+
+			const result = response.json();
+			expect(result.errors?.[0]?.extensions?.code).toBe("unexpected");
+
+			// Verify deletion
+			const user = server.drizzleClient.query.usersTable.findFirst({
+				where: eq(usersTable.emailAddress, email),
+			});
+
+			expect(user).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (harsh04044) [reply test/graphql/types/Mutation/createUser.test.ts]: this is fixed

**Comment by harsh04044** [comment]
@coderabbitai Full review

**Comment by harsh04044** [comment]
@coderabbitai Full review

**Comment by harsh04044** [comment]
@coderabbitai Full review

**Comment by harsh04044** [comment]
@coderabbitai Post your analysis as comment

**Comment by harsh04044** [comment]
@coderabbitai Full review

**Comment by harsh04044** [comment]
@coderabbitai Full review

**Comment by palisadoes** [comment]
Please ask code rabbit to do a full review of this PR and post the conclusion in an easy to read format.
